modules/oe_translation_cdt/tests/src/Kernel/XmlFormatterTest.php

```diff
@@ -0,0 +1,256 @@
+    $field_storage->save();
+
+    $field_definition = [
+      'field_storage' => $field_storage,
+      'bundle' => 'oe_demo_translatable_page',
+    ];
+    $field = FieldConfig::create($field_definition);
+    $field->save();
+
+    // Mark the test entity reference field as embeddable.
+    $this->config('oe_translation.settings')
+      ->set('translation_source_embedded_fields', [
+        'node' => [
+          'ott_content_reference' => TRUE,
+        ],
+      ])
+      ->save();
+
+    // Create a format for the content.
+    FilterFormat::create([
+      'format' => 'html',
+      'name' => 'Html',
+      'weight' => 1,
+      'filters' => [],
+    ])->save();
+
+    // Create a node to be referenced.
+    $referenced_node = Node::create([
+      'type' => 'oe_demo_translatable_page',
+      'title' => 'Referenced node',
+    ]);
+    $referenced_node->save();
+
+    // Create a node to format.
+    /** @var \Drupal\node\NodeInterface $node */
+    $node = Node::create([
+      'type' => 'oe_demo_translatable_page',
+      'title' => 'English title',
+      'ott_content_reference' => $referenced_node->id(),
+      'translatable_text_field' => [
+        'value' => '<h1>This is a heading</h1><p>This is a paragraph</p>',
+        'format' => 'html',
+      ],
+      'translatable_text_field_plain' => [
+        'value' => 'plain text field value',
+        'format' => 'plain_text',
+      ],
+    ]);
+    $node->save();
+
+    // Create a translation request with the node data.
+    $this->request = TranslationRequestCdt::create([
+      'bundle' => 'cdt',
+      'source_language_code' => $node->language()->getId(),
+      'target_languages' => [
+        'langcode' => 'fr',
+        'status' => 'Requested',
+      ],
+      'translator_provider' => 'cdt',
+    ]);
+
+    $this->request->setContentEntity($node);
+    $data = $this->container->get('oe_translation.translation_source_manager')->extractData($node->getUntranslated());
+    $this->request->setData($data);
+    $this->request->save();
+
+    $this->xml = (string) file_get_contents(\Drupal::service('extension.path.resolver')->getPath('module', 'oe_translation_cdt') . '/tests/fixtures/cdt-source-file.xml');
+    $this->xml = (string) str_replace('@transaction_identifier', (string) $this->request->id(), $this->xml);
+    $this->xml = (string) str_replace('@drupal_version', \Drupal::VERSION, $this->xml);
+    $this->xml = (string) str_replace('@module_version', InstalledVersions::getPrettyVersion('openeuropa/oe_translation'), $this->xml);
+    $this->xml = (string) str_replace('@node_id', (string) $this->request->getContentEntity()?->id(), $this->xml);
+  }
+
+  /**
+   * Test the XML content formatter export.
+   */
+  public function testXmlContentExporter(): void {
+    $time_prophecy = $this->prophesize(Time::class);
+    $time_prophecy->getCurrentTime()->willReturn(1710935729);
+
+    $formatter = new XmlFormatter(
```

Comment:
cant put my finger on it but it looks a bit strange, was expecting to see this called through the service, maybe give this a try `$this->container->set('datetime.time', $time_prophecy->reveal());`

Reply:
You're right, it made sense when I had more prophecies here, at the beginning of the ticket.